src/pretix/base/ticketoutput.py

```diff
@@ -161,3 +161,47 @@ def download_button_icon(self) -> str:
         The Font Awesome icon on the download button in the frontend.
         """
         return 'fa-download'
+
+    @property
+    def preview_allowed(self) -> bool:
+        """
+        By default, the ``generate()`` method is called for generating a preview in the pretix backend.
+        In case your plugin cannot generate previews for any reason, you can manually disable it here.
+        """
+        return True
+
+    @property
+    def is_downloadable(self) -> bool:
+        """
+        Returns whether or whether the output of this plugin can be downloaded and/or attached to an email.
+        By default, this is setting defaults to ``True`` for backwards compatibility.
+        """
+        return True
+
+    @property
+    def download_action(self) -> str:
```

Comment:
I can think of usecases where JavaScript is not required and where one would just want to return the URL to an external resource here. In that case, designing this as a ``@property`` will not be very helpful, as that URL will likely depend on the details. A more flexible signature could be

``
def download_action(self, order: Order, position: OrderPosition) -> Optional[str]
``